src/core/mod.rs

```diff
@@ -124,31 +158,8 @@ impl BPF {
         })
     }
 
-    // 0.11.0 changes the API for bpf_module_create_c_from_string()
-    #[cfg(any(
-        feature = "v0_11_0",
-        feature = "v0_12_0",
-        feature = "v0_13_0",
-        feature = "v0_14_0",
-        feature = "v0_15_0",
-        not(feature = "specific"),
-    ))]
-    /// `code` is a string containing C code. See https://github.com/iovisor/bcc for examples
-    pub fn new(code: &str) -> Result<BPF, BccError> {
-        let cs = CString::new(code)?;
-        let ptr = unsafe {
-            bpf_module_create_c_from_string(
-                cs.as_ptr(),
-                2,
-                ptr::null_mut(),
-                0,
-                true,
-                ptr::null_mut(),
-            )
-        };
-        if ptr.is_null() {
-            return Err(BccError::Compilation);
-        }
+    pub fn new_with_cflags(code: &str, cflags: Vec<&str>) -> Result<BPF, BccError> {
```

Comment:
As we might want to add more arguments soon-ish, maybe an arguments struct could be better? Unsure how _Rustic_ this is! :smile: